### PR TITLE
Feature/build screenshots on ci

### PR DESCRIPTION
**Pull request by loremattei**
This PR makes it possible to build the screenshots on CI. The PR itself is pretty big, but it's the final result of a trial project, so the various parts have been already reviewed. 

To test:
- The job can be started with this cURL call: 
```
curl -u <your_token>: -X POST --header "Content-Type: application/json" -d '{
 "branch": "feature/screenshots-phase1",
 "parameters": {
   "generate_screenshots": true
 }
}' https://circleci.com/api/v2/project/github/wordpress-mobile/WordPress-Android/pipelinegeorge:WordPress-Android
```

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by AliSoftware** [comment]
@loremattei Actually, I've included a lane to trigger the job in my changes. So to test it you should directly use this instead of your `curl` call:
```
bundle exec fastlane trigger_screenshots token:<yourtoken> branch:feature/build-screenshots-on-ci
```

File: .circleci/config.yml [hunk 1]
```diff
@@ -342,9 +345,122 @@ jobs:
...
+          apk-path: WordPress/build/outputs/apk/vanilla/debug/org.wordpress.android-vanilla-debug.apk
+          test-apk-path: WordPress/build/outputs/apk/androidTest/vanilla/debug/org.wordpress.android-vanilla-debug-androidTest.apk
+          test-targets: package org.wordpress.android.ui.screenshots
+          # NOTE: When updating this list of devices, ensure to also update the `phone:` and `tenInch:` parameters in "Download Raw Screenshots from Google Storage"
+          devices: |
+            model=blueline,version=28,orientation=portrait
+            model=gts3lltevzw,version=28,orientation=default
+          # NOTE: When updating this list of locales, ensure to also update the `promo_config:` keys accordingly in the `SUPPORTED_LOCALES` in `fastlane/Fastfile`
+          # locales: ar de_DE en_US es_ES fr_FR iw_IL in it_IT ja_JP ko_KR nl_NL pl_PL pt_BR ru_RU sr sv_SE th tr_TR vi zh_CN zh_TW
+          locales: ar de_DE en_US es_ES fr_FR iw_IL in it_IT ja_JP ko_KR nl_NL pl_PL pt_BR ru_RU sr sv_SE th tr_TR vi zh_CN zh_TW
```
Comment (AliSoftware) [inline .circleci/config.yml:392]: Leftover from when you re-enabled all locales from my limited list of 2 I guess; no need to keep the commented line now that it's the same as the uncommented one.
```suggestion
          locales: ar de_DE en_US es_ES fr_FR iw_IL in it_IT ja_JP ko_KR nl_NL pl_PL pt_BR ru_RU sr sv_SE th tr_TR vi zh_CN zh_TW
```

File: fastlane/Fastfile [hunk 2]
```diff
@@ -1,27 +1,29 @@
...
-  { glotpress: "fr", google_play: "fr-FR",  promo_config: {} },
-  { glotpress: "he", google_play: "iw-IL",  promo_config: {} },
-  { glotpress: "id", google_play: "id",  promo_config: {} },
-  { glotpress: "it", google_play: "it-IT",  promo_config: {} },
-  { glotpress: "ja", google_play: "ja-JP",  promo_config: {} },
-  { glotpress: "ko", google_play: "ko-KR",  promo_config: {} },
-  { glotpress: "nl", google_play: "nl-NL",  promo_config: {} },
-  { glotpress: "pl", google_play: "pl-PL",  promo_config: {} },
+  { glotpress: "es",    google_play: "es-ES",  promo_config: {} },
+  { glotpress: "fr",    google_play: "fr-CA",  promo_config: {} },
```
Comment (AliSoftware) [inline fastlane/Fastfile]: This one is not part of [the list of `locales:` parameter from circleci config](https://github.com/wordpress-mobile/WordPress-Android/pull/12869/files#diff-1d37e48f9ceff6d8030570cd36286a61R392) and should thus be excluded from promo screenshots
```suggestion
  { glotpress: "fr",    google_play: "fr-CA",  promo_config: false },
```

(Alternatively, maybe we could remove that line entirely to only keep `fr-FR`?)

**Review by AliSoftware — CHANGES_REQUESTED** [review]
Not sure if we should just disable `fr-CA` for `promo_config` or remove the lane entirely, but keeping it will make the job fail, not finding raw screenshots for that locale when trying to generate the corresponding promo screenshots.

**Comment by oguzkocer** [comment]
Hey @loremattei @AliSoftware! I probably won't get a chance to review this until my first full day working which is on Saturday. That is unless its review is urgent, so please let me know if you want me to prioritize it.

**Comment by AliSoftware** [comment]
Note: for reference, merging this PR should close #12601
_(I don't have the permissions to edit the PR and link it to the issue)_

**Review by jkmassel — APPROVED** [review]
This generally worked quite well!

The screenshots were captured, processed and composited. I have a few next steps that will need to be addressed before we can use it in production. Given the size of this PR already, I don't see why it can't be a separate one.

4-check-whats-happening-in-real-time
There seems to be something wrong with the opacity layer when compositing the image. This may be related to the ImageMagick 7 upgrade.

Also note in the above screenshot that the font isn't correct.

3-track-what-your-visitors-love
On tablet sizes, the screenshot image is stretched in the above screenshot


Also, somewhat weirdly; [this page shows success](https://app.circleci.com/pipelines/github/wordpress-mobile/WordPress-Android/15965/workflows/9e32160f-d1ea-470f-b760-7b690bd989fe) across the board, while [this page shows](https://app.circleci.com/pipelines/github/wordpress-mobile/WordPress-Android/15965/workflows/9e32160f-d1ea-470f-b760-7b690bd989fe/jobs/71017/tests) that 3 tests failed. This is probably related to the `fr-CA` locale that @AliSoftware mentioned earlier.

Again, I don't think these are blockers – this is generally a very good PR, but we should come back to it soon so that we can start using them soon :)

**Comment by AliSoftware** [comment]
> Also note in the above screenshot that the font isn't correct.

This might be because the font referenced in `fastlane/playstoreres/assets/*.css` are missing and will need to be installed. @oguzkocer mentioned that case in a post earlier this month.

> On tablet sizes, the screenshot image is stretched in the above screenshot

I'm wondering if I used the right orientation for the tablet device in `.circleci/config.yml`. It might be capturing raw screenshots in portrait, while the promo screenshots are rendering them I  landscape. If that's indeed the issue, could be a fix we apply directly in this PR

File: .circleci/config.yml [hunk 3]
```diff
@@ -342,9 +345,122 @@ jobs:
...
+      - android/firebase-test:
+          key-file: .circleci/.firebase.secrets.json
+          type: instrumentation
+          apk-path: WordPress/build/outputs/apk/vanilla/debug/org.wordpress.android-vanilla-debug.apk
+          test-apk-path: WordPress/build/outputs/apk/androidTest/vanilla/debug/org.wordpress.android-vanilla-debug-androidTest.apk
+          test-targets: package org.wordpress.android.ui.screenshots
+          # NOTE: When updating this list of devices, ensure to also update the `phone:` and `tenInch:` parameters in "Download Raw Screenshots from Google Storage"
+          devices: |
+            model=blueline,version=28,orientation=portrait
+            model=gts3lltevzw,version=28,orientation=default
```
Comment (AliSoftware) [inline .circleci/config.yml:389]: After seeing @jkmassel's comment on screenshots being stretched when rendered in promo material, we might want to enforce the right orientation here

```suggestion
            model=gts3lltevzw,version=28,orientation=landscape
```

**Comment by loremattei** [comment]
I agree with @jkmassel that this is already quite a big PR. 
@rachelmcr since you'll work on updating the screenshots soon, you may have a chance to fix the issues above.

**Comment by loremattei** [comment]
Thanks @AliSoftware for this work!

**Comment by oguzkocer** [comment]
Just noting here that I had a quick look through the changes and it looked great. Thank you for this work @AliSoftware! 🥳

**Pull request merged**